src/test/java/org/mockitousage/junitrule/JUnitTestRuleIntegratesWithRuleChainTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2020 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockitousage.junitrule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runners.model.Statement;
+import org.mockito.Mock;
+import org.mockito.internal.util.MockUtil;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.quality.Strictness;
+import org.mockitousage.IMethods;
+import org.mockitousage.junitrule.MutableStrictJUnitTestRuleTest.StrictByDefault;
+import org.mockitoutil.JUnitResultAssert;
+
+public class JUnitTestRuleIntegratesWithRuleChainTest {
+
+    JUnitCore runner = new JUnitCore();
+
+    @Test public void rule_can_be_changed_to_strict() {
+        //when
+        Result result = runner.run(MutableStrictJUnitTestRuleTest.LenientByDefault.class);
+
+        //then
+        JUnitResultAssert.assertThat(result)
+            .succeeds(1)
```

Comment:
take a look at other methods on JUnitResultAssert - we can make specific assertions on the names of failed tests, etc.

Reply:
This is taken from `MutableStrictJUnitTestRuleTest`.